modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/getDataSetRelatedResourcesURL.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+export default function getDataSetRelatedResourcesURL({
+	dataSetId,
+	params,
+	relatedResourceId,
+	relationship,
+}: {
+	dataSetId?: string;
+	params?: Record<string, string>;
+	relatedResourceId?: string;
+	relationship?: string;
+}): string {
+	let apiURL = '/o/data-set-admin/data-sets';
+
+	if (dataSetId && relationship) {
+		apiURL += `/${dataSetId}/${relationship}`;
+
+		if (relatedResourceId) {
+			apiURL += `/${relatedResourceId}`;
+		}
+	}
+
+	if (params) {
+		apiURL += `?${new URLSearchParams(params).toString()}`;
+	}
+
+	return apiURL;
+}
```

Comment:
This is the central part of PR. Some URL need to be converted to this pattern, some still work with old.